### PR TITLE
Fix lint warnings in JavaScript tests

### PR DESCRIPTION
**Pull request by cwdavies**
Fix lint warnings with:
- camelcase
- max-len
- no-console
- no-shadow
- no-unused-vars
- prefer-const

Examples of warning:
- This line has a length of N. Maximum allowed is 80  max-len
- 'chartMenu' is already declared in the upper scope  no-shadow
- 'chartMenu' is assigned a value but never used      no-unused-vars
- 'discountEl' is never reassigned. Use 'const' instead  prefer-const
- Unexpected console statement  no-console
- Identifier 'counseling_agencies' is not in camel case  camelcase
- Identifier 'html_url' is not in camel case  camelcase

## Additions

- eslint-disable-next-line no-unused-vars
- eslint-disable-next-line no-console
- eslint-disable-next-line camelcase

## Removals

- unused variables

## Changes

- split lines longer than 80 characters onto multiple lines
- variable use of const

## How to test this PR

1. gulp lint:tests
2. gulp test:unit --ci --headless
3. Verify no tests start to fail after fixing lint warning

## Notes and todos

Fix lint warnings for:
- Unexpected 'todo' comment  no-warning-comments

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: test/unit_tests/apps/regulations3k/js/recent-notices-spec.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ describe( 'The Regs3K search page', () => {
     mockXHR.onreadystatechange();
 
     setTimeout( () => {
-      expect( console.error ).toBeCalled();
+      // expect( console.error ).toBeCalled();
```
Comment (anselmbradford) [inline test/unit_tests/apps/regulations3k/js/recent-notices-spec.js:116]: This test is running but not testing anything anymore. The test is testing `should fail to load recent notices`, so is asserting that `console.error` is thrown by the underlying code. The options we have here that I can think of:
- Revert this change.
- Change the underlying code to not require `console.error` (this logs a console message in production, do we want that?).
- Adjust the global eslint rule to only flag console.log, not console.error and console.warn (https://eslint.org/docs/rules/no-console#options)

File: test/unit_tests/apps/regulations3k/js/search-spec.js [hunk 2]
```diff
@@ -142,7 +142,7 @@ describe( 'The Regs3K search page', () => {
 
     simulateEvent( 'click', clearIcon );
     setTimeout( () => {
-      expect( console.error ).toBeCalled();
+      // expect( console.error ).toBeCalled();
```
Comment (anselmbradford) [inline test/unit_tests/apps/regulations3k/js/search-spec.js:145]: See other console.error comment

File: test/unit_tests/apps/teachers-digital-platform/js/expandable-facets-spec.js [hunk 3]
```diff
@@ -66,9 +66,9 @@ describe( 'Expandable facets', () => {
     ef = document.querySelector( '.o-expandable-facets' );
     expandableFacet = new ExpandableFacets( ef );
     expandableFacet.init();
-    efLabel = document.querySelector( '.o-expandable-facets_checkbox ~ .a-label' );
+    // efLabel = document.querySelector( '.o-expandable-facets_checkbox ~ .a-label' );
     efTarget = document.querySelector( '.o-expandable-facets_target' );
-    efHeader = document.querySelector( '.o-expandable_header' );
+    // efHeader = document.querySelector( '.o-expandable_header' );
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/expandable-facets-spec.js:71]: Commented out because they are unused? Coverage for this file is at 94.44%. We could look if these are worth including in additional tests, or they could be deleted.

File: test/unit_tests/apps/teachers-digital-platform/js/search-spec.js [hunk 4]
```diff
@@ -281,7 +281,7 @@ describe( 'The TDP search page', () => {
 
     simulateEvent( 'click', clearIcon );
     setTimeout( () => {
-      expect( console.error ).toBeCalled();
+      // expect( console.error ).toBeCalled();
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/search-spec.js:284]: See other console.error comment

File: test/unit_tests/apps/teachers-digital-platform/js/search-spec.js [hunk 5]
```diff
@@ -247,7 +247,7 @@ describe( 'The TDP search page', () => {
     };
     global.XMLHttpRequest = jest.fn( () => mockXHR );
     const parentCheckbox = document.querySelector( '#topic--earn' );
-    const form = document.querySelector( 'form#filter-form' );
+    // const form = document.querySelector( 'form#filter-form' );
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/search-spec.js:250]: Seems like it could be deleted?

File: test/unit_tests/apps/youth-employment-success/js/reducers/route-option-reducer-spec.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
-import createRoute from '../../../../../../cfgov/unprocessed/apps/youth-employment-success/js/models/route';
+import { createRoute
+} from '../../../../../../cfgov/unprocessed/apps/youth-employment-success/js/models/route';
```
Comment (anselmbradford) [inline test/unit_tests/apps/youth-employment-success/js/reducers/route-option-reducer-spec.js:2]: I'd probably make this:

```
import { 
  createRoute
} from '../../../../../../cfgov/unprocessed/apps/youth-employment-success/js/models/route';
```

So it follows how we generally format object literals.

File: test/unit_tests/js/modules/util/add-email-popup-spec.js [hunk 7]
```diff
@@ -57,7 +55,6 @@ describe( 'add-email-popup', () => {
   } );
 
   it( 'should hide the email popup initially', () => {
-    const addEmailPopup = require( BASE_JS_PATH + '/add-email-popup' );
```
Comment (anselmbradford) [inline test/unit_tests/js/modules/util/add-email-popup-spec.js:60]: This is removing the file that is being tested. With this change the spec won't be testing anything and the coverage will be 0% for that file.

File: test/unit_tests/js/modules/util/breakpoint-state-spec.js [hunk 8]
```diff
@@ -36,32 +34,41 @@ describe( 'breakpoint-state', () => {
   describe( '.getBreakpointState()', () => {
     it( 'should correctly return bpXS state', () => {
       windowResizeTo( 300 );
-      expect( getBreakpointState() ).toStrictEqual( { bpLG: false, bpMED: false, bpSM: false, bpXL: false, bpXS: true } );
+      expect( getBreakpointState() ).toStrictEqual(
+        { bpLG: false, bpMED: false, bpSM: false, bpXL: false, bpXS: true }
```
Comment (anselmbradford) [inline test/unit_tests/js/modules/util/breakpoint-state-spec.js:38]: Could stack these, but it's up to your preference.
```
{ 
  bpLG: false, 
  bpMED: false, 
  bpSM: false, 
  bpXL: false, 
  bpXS: true 
}
```

Comment (anselmbradford) [reply test/unit_tests/js/modules/util/breakpoint-state-spec.js:38]: Same for additional occurrences in this file.

**Review by anselmbradford — DISMISSED** [review]

File: test/unit_tests/js/modules/util/breakpoint-state-spec.js (hunk 8 above)
Comment (cwdavies) [reply test/unit_tests/js/modules/util/breakpoint-state-spec.js]: I agree and put each breakpoint state on separate lines

File: test/unit_tests/apps/youth-employment-success/js/reducers/route-option-reducer-spec.js (hunk 6 above)
Comment (cwdavies) [reply test/unit_tests/apps/youth-employment-success/js/reducers/route-option-reducer-spec.js:3]: I made the change and put import of createRoute on a separate line

File: test/unit_tests/js/modules/util/add-email-popup-spec.js (hunk 7 above)
Comment (cwdavies) [reply test/unit_tests/js/modules/util/add-email-popup-spec.js:60]: I fixed this lint warning with no-unused-vars by using eslint-disable-next-line before addEmailPopup

File: test/unit_tests/apps/teachers-digital-platform/js/search-spec.js (hunk 4 above)
Comment (cwdavies) [reply test/unit_tests/apps/teachers-digital-platform/js/search-spec.js]: I fixed this lint warning with no-console by using eslint-disable-next-line before console.error line

File: test/unit_tests/apps/regulations3k/js/search-spec.js (hunk 2 above)
Comment (cwdavies) [reply test/unit_tests/apps/regulations3k/js/search-spec.js]: I fixed this lint warning with no-console by using eslint-disable-next-line before console.error line

File: test/unit_tests/apps/regulations3k/js/recent-notices-spec.js (hunk 1 above)
Comment (cwdavies) [reply test/unit_tests/apps/regulations3k/js/recent-notices-spec.js]: I fixed this lint warning with no-console by using eslint-disable-next-line before each console.error line
This option was simpler than adjusting the global eslint rule to only flag `console.log`
If there are many occurrences of `console.warn` and `console.error` this the eslint rule option would be better

File: test/unit_tests/apps/teachers-digital-platform/js/expandable-facets-spec.js (hunk 3 above)
Comment (cwdavies) [reply test/unit_tests/apps/teachers-digital-platform/js/expandable-facets-spec.js]: I made an attempt to check if label or header are hidden.
@anselmbradford What is the command line to run unit_tests that would generate code coverage?

File: test/unit_tests/apps/owning-a-home/js/explore-rates/rate-checker-spec.js [hunk 9]
```diff
@@ -177,8 +178,9 @@ describe( 'explore-rates/rate-checker', () => {
     beforeEach( () => {
       document.body.innerHTML = HTML_SNIPPET;
       downPaymentDom = document.querySelector( '#down-payment' );
-      rateStructureDom = document.querySelector( '#rate-structure' );
-      armTypeDom = document.querySelector( '#arm-type' );
+
+      /* rateStructureDom = document.querySelector( '#rate-structure' );
+         armTypeDom = document.querySelector( '#arm-type' ); */
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/explore-rates/rate-checker-spec.js]: Yes, they were commented out because they were not used and probably reminders that more tests are needed.
Instead of deleting them, I made an attempt to check value of rate-structure and ARM type.
@anselmbradford What is the command line that generates code coverage?

File: test/unit_tests/apps/teachers-digital-platform/js/search-spec.js (hunk 5 above)
Comment (cwdavies) [reply test/unit_tests/apps/teachers-digital-platform/js/search-spec.js]: Yes, this is an unused variable so I deleted it.

File: test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js [hunk 10]
```diff
@@ -93,8 +95,10 @@ describe( 'Dropdown utils', () => {
   } );
 
   it( 'should select several select elements', () => {
-    document.body.innerHTML += '<select id="foo1"><option value="baz1"></option></select>';
-    document.body.innerHTML += '<select id="foo2"><option value="baz1"></option></select>';
+    document.body.innerHTML +=
+      '<select id="foo1"><option value="baz1"></option></select>';
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: Fixed

File: test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js [hunk 11]
```diff
@@ -93,8 +95,10 @@ describe( 'Dropdown utils', () => {
...
+    document.body.innerHTML +=
+      '<select id="foo2"><option value="baz1"></option></select>';
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: Fixed.

File: test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js [hunk 12]
```diff
@@ -93,8 +95,12 @@ describe( 'Dropdown utils', () => {
   } );
 
   it( 'should select several select elements', () => {
-    document.body.innerHTML += '<select id="foo1"><option value="baz1"></option></select>';
-    document.body.innerHTML += '<select id="foo2"><option value="baz1"></option></select>';
+    document.body.innerHTML += `
+<select id="foo1"><option value="baz1"></option></select>
+`;
+    document.body.innerHTML += `
+<select id="foo2"><option value="baz1"></option></select>
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: Fixed

File: test/unit_tests/apps/youth-employment-success/js/input-view-spec.js [hunk 13]
```diff
@@ -22,10 +24,12 @@ describe( 'InputView', () => {
     view = inputView( document.querySelector( 'div' ) );
     view.init();
 
-    expect( document.querySelector( 'div' ).getAttribute( 'data-js-hook' ) ).toBeTruthy();
+    expect( document.querySelector( 'div' )
+      .getAttribute( 'data-js-hook' ) ).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/youth-employment-success/js/input-view-spec.js]: Fixed

File: test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: Fixed?

[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: Fixed?

File: test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js [hunk 14]
```diff
@@ -95,8 +95,10 @@ describe( 'Dropdown utils', () => {
   } );
 
   it( 'should select several select elements', () => {
+    /* eslint-disable max-line */
```
Comment (anselmbradford) [inline test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: `max-len`?

Comment (cwdavies) [reply test/unit_tests/apps/owning-a-home/js/dropdown-utils-spec.js]: That was a typo! I change max-line to max-len

File: .eslintrc [hunk 15]
```diff
@@ -802,7 +802,6 @@ rules:
       ignoreComments: true
       ignoreTrailingComments: true
       ignoreUrls: true
-      ignoreStrings: true
```
Comment (anselmbradford) [inline .eslintrc:805]: Mentioned in chat… I would revert this change and keep `ignoreStrings: true` since that is the default, but having it here shows the behavior without needing to look it up in the docs.

Comment (cwdavies) [reply .eslintrc]: I added ignoreStrings back into the main .eslintrc. There were more max-len lint warnings when I removed ignoreStrings then when it was set to true. It seemed like the default setting was false.

File: test/.eslintrc [hunk 16]
```diff
@@ -15,3 +15,14 @@ rules:
   max-statements: 0
   no-unused-expressions: 0
   require-jsdoc: 0
+  max-len:
+    - 1
+    -
+      // https://regex101.com/r/hVU4sX/1
+      ignorePattern: '(^import\s([*])|([A-Za-z0-9]*)\s)|(\sfrom\s)'
```
Comment (Scotchester) [inline test/.eslintrc:22]: @cfpb/front-end-admins Would we want to add this `ignorePattern` for `import`s to our global `.eslintrc`?

**Comment by cwdavies** [comment]
According to https://eslint.org/docs/rules/max-len,
when `ignoreStrings: true` - lines that contain a double-quoted or single-quoted string are ignored.

With `ignoreStrings: false` - lines that contain a double-quoted or single-quoted string are **not** ignored.

If a string is longer than 80 character, as defined by `max-len` in `.eslintrc`, it should be split onto multiple lines.

@anselmbradford wanted me to make changes in `test/.eslintrc` so this would only impact strings in JavaScript tests.

The exception to this rule is when using import there will be paths that are longer than 80 characters,
so there is an `ignorePattern` defined to ignore a specific regex for import from.

**Review by ascott1 — CHANGES_REQUESTED** [review]
@cwdavies I don't think that we should be changing the linter here as it has wide reaching consequences across the code base. I'd recommend reverting the linting changes and update the tests to match the linting to work within those constraints.

We may want to consider some of these changes and update our standards accordingly, but please discuss them with @cfpb/front-end-admins (or the FEWD chat channel). If we do make changes, I'd recommend that we implement a code formatter to ensure consistency across the codebase. Anything like that should be done as a separate initiative, not as part of this change.

I'd be happy to discuss this further.

**Comment by cwdavies** [comment]
> @cwdavies I don't think that we should be changing the linter here as it has wide reaching consequences across the code base. I'd recommend reverting the linting changes and update the tests to match the linting to work within those constraints.
> 
> We may want to consider some of these changes and update our standards accordingly, but please discuss them with @cfpb/front-end-admins (or the FEWD chat channel). If we do make changes, I'd recommend that we implement a code formatter to ensure consistency across the codebase. Anything like that should be done as a separate initiative, not as part of this change.
> 
> I'd be happy to discuss this further.

@ascott1 Thanks for your feedback Adam. I will remove changes I made to the linter and we can get this PR closed.
I like your idea of implementing a code formatter as a separate initiative.
I would suggest adding `ignorePattern` into the FEWD standards - this was actually a suggestion made by @anselmbradford

File: test/.eslintrc (hunk 16 above)
Comment (cwdavies) [reply test/.eslintrc]: Removed `ignorePattern` and `ignoreStrings` from `.eslintrc` based on feedback from @ascott1

**Comment by ascott1** [comment]
> > @cwdavies I don't think that we should be changing the linter here as it has wide reaching consequences across the code base. I'd recommend reverting the linting changes and update the tests to match the linting to work within those constraints.
> > We may want to consider some of these changes and update our standards accordingly, but please discuss them with @cfpb/front-end-admins (or the FEWD chat channel). If we do make changes, I'd recommend that we implement a code formatter to ensure consistency across the codebase. Anything like that should be done as a separate initiative, not as part of this change.
> > I'd be happy to discuss this further.
> 
> @ascott1 Thanks for your feedback Adam. I will remove changes I made to the linter and we can get this PR closed.
> I like your idea of implementing a code formatter as a separate initiative.
> I would suggest adding `ignorePattern` into the FEWD standards - this was actually a suggestion made by @anselmbradford

That all sounds good, thank you @cwdavies!

File: test/unit_tests/apps/youth-employment-success/js/input-view-spec.js [hunk 17]
```diff
@@ -22,10 +22,13 @@ describe( 'InputView', () => {
...
     view.init();
 
-    expect( document.querySelector( 'div' ).getAttribute( 'data-js-hook' ) ).toBeTruthy();
+    expect( document.querySelector( 'div' )
+      .getAttribute( 'data-js-hook' )
+    ).toBeTruthy();
   } );
 
-  it( 'throws an error when an <input> node matching the `type` prop cannot be found', () => {
+  it( 'throws an error when an <input> node matching ' +
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/unit_tests/apps/youth-employment-success/js/input-view-spec.js:30]: Done

File: test/unit_tests/apps/ccdb-landing-map/js/Chart-spec.js [hunk 18]
```diff
@@ -1,5 +1,7 @@
 import 'regenerator-runtime/runtime';
-import Chart from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/Chart';
+import
+Chart
+  from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/Chart';
```
Comment (Scotchester) [inline test/unit_tests/apps/ccdb-landing-map/js/Chart-spec.js:4]:
```suggestion
import {
  Chart
} from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/Chart';
```

**Review by Scotchester — CHANGES_REQUESTED** [review]
This is looking very close! I just have one request for every place in which imports were broken across multiple lines. The way in which you have done that makes it more difficult to read, in my opinion. I suggest using the curly brace formatting that is seen already in some existing imports (even with only one item being imported) would be more clear. Could you please apply that to all of the import changes in this PR?

File: test/unit_tests/apps/ccdb-landing-map/js/Chart-spec.js (hunk 18 above)
Comment (cwdavies) [reply test/unit_tests/apps/ccdb-landing-map/js/Chart-spec.js:4]: I am updating all import lines to use curly brace formatting, even if there is only one item, for two reasons:
1. Consistency in that all imports have the same formatting
2. Import lines should not be difficult to read

**Comment by Scotchester** [comment]
Linting is looking good, but something about these changes is breaking a lot of unit tests. @cwdavies could you take a look at that?

**Comment by cwdavies** [comment]
> Linting is looking good, but something about these changes is breaking a lot of unit tests. @cwdavies could you take a look at that?

The tests broke due to the adding of curly brace formatting when there is just one module imported.
I have removed the curly brace formatting and now the tests pass.
It appears that the curly brace formatting works when there are multiple modules imported.

**Comment by Scotchester** [comment]
> The tests broke due to the adding of curly brace formatting when there is just one module imported.
> I have removed the curly brace formatting and now the tests pass.
> It appears that the curly brace formatting works when there are multiple modules imported.

Wow! That is surprising and frustrating. Sorry for sending you on a goose chase.

**Review by Scotchester — APPROVED** [review]
I confirm that the unit tests pass again and that all test lint warnings are resolved except for ones about TODO comments.

File: test/unit_tests/apps/ccdb-landing-map/js/landing-map-spec.js [hunk 19]
```diff
@@ -1,7 +1,7 @@
 import 'regenerator-runtime/runtime';
 import * as sinon from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/node_modules/sinon';
-import Chart from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/Chart.js';
-import landingMap from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/landing-map.js';
+import Chart from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/Chart';
+import landingMap from '../../../../../cfgov/unprocessed/apps/ccdb-landing-map/js/landing-map';
```
Comment (anselmbradford) [inline test/unit_tests/apps/ccdb-landing-map/js/landing-map-spec.js:4]: I tend to leave the `.js` so if you search for the filename its references will come up in the search results, but we do it both ways so whichever is preferred in this context is fine.

Comment (cwdavies) [reply test/unit_tests/apps/ccdb-landing-map/js/landing-map-spec.js]: Done. I noticed some imports have that and some don't have the js file extension.
I feel that it is better to have the `.js` so it is easier to search.
